client/src/featureform/register.py

```diff
@@ -1218,8 +1222,15 @@ def __call__(self, fn):
         if not isinstance(self.inputs, list):
             raise ValueError("Dataframe transformation inputs must be a list")
 
+        # check that input isn't self referencing
         for nv in self.inputs:
-            if self.name is nv[0] and self.variant is nv[1]:
+            if isinstance(
+                nv, tuple
+            ):  # TODO all that should be called here is name_variant()
```

Comment:
with the above changes, do we still need to call `ininstance()`, can we just rely on `nv.name_variant()`?